integration_tests/src/main/python/spark_session.py

```diff
@@ -189,3 +189,13 @@ def get_java_major_version():
     elif dash_pos != -1:
         ver = ver[0:dash_pos]
     return int(ver)
+
+def get_jvm_charset():
+    sc = _spark.sparkContext
+    return str(sc._jvm.java.nio.charset.Charset.defaultCharset())
+
+def is_jvm_charset_utf8():
+    return get_jvm_charset() == 'UTF-8'
+
+def is_jvm_charset_not_utf8():
+    return get_jvm_charset() != 'UTF-8'
```

Comment:
Nit: This is extra code to maintain that seems unnecessary.  I'm assuming this was created for readability, but if we want to check for not is_jvm_charset_utf8(), then we would type exactly that:
```python
  if not is_jvm_charset_utf8():
    ...
```

IMHO this is quite readable and precludes the need to maintain the separate method.

Reply:
cleaned up the code a bit based on this feedback in most recent commit